go-controller/pkg/ovn/master.go

```diff
@@ -1022,3 +1029,101 @@ func (oc *DefaultNetworkController) deleteHoNodeEvent(node *kapi.Node) error {
 	}
 	return nil
 }
+
+// addIPToHostNetworkNamespaceAddrsSet adds management port IP and node's
+// gateway-router-lrp-ifaddr to address_set created for HostNetworkNamespace.
+// This function gets called from both addUpdateLocalNodeEvent & addUpdateRemoteNodeEvent
+// to add IPs to address_set for both local and remote zone nodes.
+func (oc *DefaultNetworkController) addIPToHostNetworkNamespaceAddrsSet(node *kapi.Node) error {
```

Comment:
too much code duplication between this function and the one below.
except:
`nsInfo.addressSet.AddIPs(hostNetworkPolicyIPs); err != nil` and `nsInfo.addressSet.DeleteIPs(hostNetworkPolicyIPs); err != nil` everything else is copy, a bunch of stuff can be shared

Reply:
see if you can leverage: `getAllHostNamespaceAddresses`  or do something similar to that like `getHostNamespaceAddressesForNode` but per node and then combine that into getAllHostNamespaceAddresses which calls it out..?

Reply:
also why are the bits inside `ensureNodeLogicalNetwork` not deleted? I expected the bits there to go away.. we don't need them right?

Reply:
getAllHostNamespaceAddresses seems appropriate to reduce code duplication. Thanks for pointing this out.

Reply:
Will remove ensureNodeLogicalNetwork entirely in next push as that function just used to add IPs to HostnetworkNamespace specific address_set and now we are putting this logic in AddResource/UpdateResurce.